plugins/assets/s3/src/index.ts

```diff
@@ -41,7 +41,7 @@ class S3Assets extends Assets {
     }
     const { buffer, filename } = await this.analyze(url, file)
     const s3Key = `${this.config.pathPrefix}${filename}`
-    const finalUrl = `${this.config.publicUrl}${filename}`
+    const finalUrl = `${this.config.publicUrl}${encodeURIComponent(filename)}`
```

Comment:
Is slash expected as a suffix for `publicUrl`?

Reply:
This is designed to match the same behavior as `pathPrefix`.